### PR TITLE
First part of the migration to era for phase2 tracking

### PR DESCRIPTION
**Pull request by ebrondol**
This PR migrates the Phase2 iterative tracking customization from `phase2TkTilted/phase2TkFlat` to `trackingPhase2PU140` era. The transition has been done trying to not have any difference respect to the customization function. The comparison with before and after era does not show any significant difference in the MTV plots using TTbar sample for both [flat](http://ebrondol.web.cern.ch/ebrondol/Phase2Tracking/MovingToEra/13July/TTbar_Flat/plots_ootb/effandfake1.pdf) and [tilted](http://ebrondol.web.cern.ch/ebrondol/Phase2Tracking/MovingToEra/13July/TTbar_Tilted/plots_ootb/effandfake1.pdf) geometry.
Next steps are:
- Integrating `duplicateMerge` in  the Validation part (there is a FIXME [here](https://github.com/cms-sw/cmssw/compare/CMSSW_8_1_X...ebrondol:trackingPhase2_eraMigration_81X#diff-aadf335934a95c465835668d66744a38R37))
- Moving `layerList` to `PixelLayerTriplets.layerList` (there is a FIXME [here](https://github.com/cms-sw/cmssw/compare/CMSSW_8_1_X...ebrondol:trackingPhase2_eraMigration_81X#diff-f008ec4cef4654fe8e8e19358685f3cdR27))
- Migrate rest of `phase2TkTilted/phase2TkFlat` to eras

No difference is expected also for phase0/1 iterative tracking.

Thanks to @makortel for the precious help.
Informing @VinInn @rovere @delaere @boudoul 
@kpedro88 if possible in pre9.

### CODE REVIEW

**Comment by VinInn** [comment]
@cmsbuild , please test

File: SLHCUpgradeSimulations/Configuration/python/phase2TkFlat.py [hunk 1]
```diff
@@ -1,4 +1,5 @@
 import FWCore.ParameterSet.Config as cms
+from Configuration.StandardSequences.Eras import eras
```
Comment (kpedro88) [inline SLHCUpgradeSimulations/Configuration/python/phase2TkFlat.py]: Why is this import needed?

File: SLHCUpgradeSimulations/Configuration/python/phase2TkTilted.py [hunk 2]
```diff
@@ -1,4 +1,5 @@
 import FWCore.ParameterSet.Config as cms
+from Configuration.StandardSequences.Eras import eras
```
Comment (kpedro88) [inline SLHCUpgradeSimulations/Configuration/python/phase2TkTilted.py]: Why is this import needed?

**Comment by kpedro88** [comment]
@ebrondol at a glance, this looks good to me. Hopefully @slava77, @cvuosalo, @civanch can review quickly. I'm not sure what the intended date for pre9 is.

**Comment by slava77** [comment]
On 7/13/16 5:28 AM, Kevin Pedro wrote:
> @ebrondol <https://github.com/ebrondol> at a glance, this looks good to
> me. Hopefully @slava77 <https://github.com/slava77>, @cvuosalo
> <https://github.com/cvuosalo>, @civanch <https://github.com/civanch> can
> review quickly. I'm not sure what the intended date for pre9 is.
> 

We are somewhat on schedule
https://twiki.cern.ch/twiki/bin/viewauth/CMS/CMSSW_8_1_0
So, this PR is technically late and will go in on best effort.

**Comment by ebrondol** [comment]
Hi @slava77 yes, sure. It would be better to have it, but there is no strong need.

**Comment by kpedro88** [comment]
We have to wait for #15188 to run the tests.

**Comment by ebrondol** [comment]
Ok, tnks for the update.

File: SLHCUpgradeSimulations/Configuration/python/phase2TkFlat.py (hunk 1 above)
Comment (ebrondol) [inline SLHCUpgradeSimulations/Configuration/python/phase2TkFlat.py]: It is not used now but I would like to keep it because every time that I I move part of the custom file, I use 

    if not eras.trackingPhase2PU140.isChosen():

to comment out the parts that I move.

File: SLHCUpgradeSimulations/Configuration/python/phase2TkTilted.py (hunk 2 above)
Comment (ebrondol) [inline SLHCUpgradeSimulations/Configuration/python/phase2TkTilted.py]: Same https://github.com/cms-sw/cmssw/pull/15183#discussion_r70634532

**Comment by slava77** [comment]
@cmsbuild please test

**Comment by civanch** [comment]
+1

**Comment by cvuosalo** [comment]
@ebrondol: Jenkins results are showing differences for workflow 10624.0_TTbar13TeV2023. Conversion step tracks seem to disappear:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_8_1_X_2016-07-13-1100+15183/14932/validateJR/all_OldVSNew_TTbar13TeV2023wf10624p0/

Also, seeding changes, as shown in one example below. I understood you to say that this PR should not create significant changes in 2023 quantities. Is there some error?


seeding

**Comment by cvuosalo** [comment]
@ebrondol: This PR also causes a new error to appear in the step3 log for workflow 10624.0_TTbar13TeV2023:

```
%MSG-e PhotonConversionFinderFromTracks:   PhotonConversionTrajectorySeedProducerFromSingleLeg:photonConvTrajSeedFromSingleLeg  13-Jul-2016 23:13:57 CEST Run: 1 Event: 1
Error! Can't get the product primary Vertex Collection InputTag:  label = firstStepPrimaryVertices, instance = 
```

**Comment by makortel** [comment]
@cvuosalo 
> Also, seeding changes, as shown in one example below. I understood you to say that this PR should not create significant changes in 2023 quantities. Is there some error?

The seeding layer set passed to MultiTrackValidator is essentially garbage for phase2 before this PR. The code constructing the set (https://github.com/cms-sw/cmssw/blob/CMSSW_8_1_X/Validation/RecoTrack/python/TrackValidation_cff.py#L217) assumes the seeding layers are specified in RECO before the code gets run, and that is the case with eras, but not with customize function. I.e. this is a "feature" of monitoring, not RECO.

If there were really changes in seeding, those would show up in all over the place.

**Comment by makortel** [comment]
@ebrondol I believe the disappearance of conversions and the error message in https://github.com/cms-sw/cmssw/pull/15183#issuecomment-232799510 are correlated (the disappearance is also visible in your plots, see e.g. [here](http://ebrondol.web.cern.ch/ebrondol/Phase2Tracking/MovingToEra/13July/TTbar_Flat/plots_conversion_conversionStep/effandfake1.pdf), we just missed it).

I think you need to add
```
eras.trackingPhase2PU140.toModify(photonConvTrajSeedFromSingleLeg, primaryVerticesTag = "pixelVertices")
```
here https://github.com/ebrondol/cmssw/blob/a800f34541a2836637191de98a53f4bdd8d5b6ce/RecoTracker/ConversionSeedGenerators/python/ConversionStep_cff.py#L370

**Comment by ebrondol** [comment]
@cvuosalo thanks you very much to spot the issue! I fix it and it should be correct now (thanks @makortel for the hint).

**Comment by slava77** [comment]
@cmsbuild please test

**Comment by kpedro88** [comment]
@cmsbuild please test

**Comment by cvuosalo** [comment]
Extended test in progress...

**Comment by cvuosalo** [comment]
+1

For #15183 3be63fbc3b01603dbc49cfc5842577b301563c5d

Part 1 of era migration for Phase 2 tracking.

The code changes are satisfactory. Jenkins tests against baseline CMSSW_8_1_X_2016-07-14-2300 show no significant differences, except for changes in DQM seeding plots that are expected, as discussed above. An extended test of upgrade workflow 10624.0_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2023tilted with 20 events against baseline CMSSW_8_1_X_2016-07-12-1600 also shows no significant differences except the expected ones in the DQM seeding plots. RECO event content shows no significant change, the expanded RECO configs show only the expected changes due to this PR, and the logs for the RECO step show no important differences.

**Comment by cvuosalo** [comment]
@civanch, @dmitrijus: Ping. Phase 2 group would like to get this PR approved promptly for 810pre9, if possible. Thanks.

**Comment by civanch** [comment]
+1
